openqaoa/workflows/optimizer.py

```diff
@@ -713,26 +713,29 @@ def optimize(self, verbose=False):
             spin_map = rqaoa.spin_mapping(problem, max_terms_and_stats)
             # Eliminate spins and redefine problem
             new_problem, spin_map = rqaoa.redefine_problem(problem, spin_map)
+            if new_problem == problem:
+                print("Eliminations lead to a total reduction of the problem.\n Increasing the cutoff to solve the smallest non-vanishing problem.")
+                break
+            else:
```

Comment:
this `else` is not needed, and I would say that without it the code is more readable, since `new_problem == problem` is a rare case

Reply:
Good point! Thanks! Done!